src/main/java/org/shanerx/tradeshop/utils/Utils.java

```diff
@@ -0,0 +1,422 @@
+	 */
+	public String colorize(String msg) {
+		msg = ChatColor.translateAlternateColorCodes('&', msg);
+		return msg;
+	}
+
+	/**
+	 * Finds the TradeShop sign linked to a chest.
+	 *
+	 * @param chest the block holding the shop's inventory. Can be a chest, a trapped chest, a dropper, a dispenser, a hopper and a shulker box (1.9+).
+	 * @return the sign.
+	 */
+	public Sign findShopSign(Block chest) {
+		ArrayList<BlockFace> faces = plugin.getListManager().getDirections();
+		Collections.reverse(faces);
+		ArrayList<BlockFace> flatFaces = new ArrayList<>(Arrays.asList(BlockFace.NORTH, BlockFace.EAST, BlockFace.SOUTH, BlockFace.WEST));
+		boolean isDouble = false;
+		BlockFace doubleSide = null;
+
+		for (BlockFace face : faces) {
+			Block relative = chest.getRelative(face);
+			if (ShopType.isShop(relative)) {
+				return (Sign) relative.getState();
+			} else if (flatFaces.contains(face) && (chest.getType().equals(Material.CHEST) || chest.getType().equals(Material.TRAPPED_CHEST))) {
+				if (relative.getType().equals(chest.getType())) {
+					isDouble = true;
+					doubleSide = face;
+				}
+			}
+		}
+
+		if (isDouble) {
+			chest = chest.getRelative(doubleSide);
+			for (BlockFace face : faces) {
+				Block relative = chest.getRelative(face);
+				if (ShopType.isShop(relative)) {
+					return (Sign) relative.getState();
+				}
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Finds the TradeShop chest, dropper, dispenser, hopper or shulker box (1.9+) linked to a sign.
+	 *
+	 * @param sign the TradeShop sign
+	 * @return the shop's inventory holder block.
+	 */
+	public Block findShopChest(Block sign) {
+		ArrayList<Material> invs = plugin.getListManager().getInventories();
+		ArrayList<BlockFace> faces = plugin.getListManager().getDirections();
+
+		for (BlockFace face : faces) {
+			Block relative = sign.getRelative(face);
+			if (relative != null && invs.contains(relative.getType())) {
+				return relative;
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Checks to see if the shop chest is findable
+	 *
+	 * @param sign the TradeShop sign
+	 * @return whether the sign is linked to a chest.
+	 */
+	public boolean checkShopChest(Block sign) {
+		return findShopChest(sign) != null;
+	}
+
+	public static Block getOtherHalfOfDoubleChest(Block chest) {
+		if (chest.getType() != Material.CHEST || chest.getType() != Material.TRAPPED_CHEST) {
+			return null;
+		}
+		ArrayList<BlockFace> flatFaces = new ArrayList<>(Arrays.asList(BlockFace.NORTH, BlockFace.EAST, BlockFace.SOUTH, BlockFace.WEST));
+
+		for (BlockFace face : flatFaces) {
+			Block adjoining = chest.getRelative(face);
+
+			if (adjoining.getType() == chest.getType()) {
+				return adjoining;
+			}
+		}
+
+		return null;
+	}
+
+	public static boolean isDoubleChest(Block chest) {
+		return getOtherHalfOfDoubleChest(chest) != null;
+	}
+
+	public void debug(String text) {
+		if (Setting.ENABLE_DEBUG.getBoolean()) {
+			plugin.getLogger().log(Level.WARNING, text);
+		}
+	}
+}
```

Comment:
Since we no longer have a massive file with every utility function thrown into it, and since it is inside the `utils` package, perhaps this can be renamed into something more meaningful.
Or we could delete the file and move the functions to other files, ie. `ShopChest`.

Reply:
that's the idea, just wasn't the biggest priority once it was trimmed down this far

Reply:
Fair enough